lib/content_encoding.c

```diff
@@ -281,8 +285,12 @@ static CURLcode deflate_do_write(struct Curl_easy *data,
     return Curl_cwriter_write(data, writer->next, type, buf, nbytes);
 
   /* Set the compressed input when this function is called */
-  z->next_in = (Bytef *) buf;
-  z->avail_in = (uInt) nbytes;
+#ifdef z_const
```

Comment:
z_const was added to zlib 13 years ago in v1.2.5.2. We currently mandate v1.2.0.4 as the minimum which is over 21 years ago. We can eliminate this conditional code by bumping the minimum.

Reply:
Sounds good to me!

Reply:
PR: #16616